fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -3458,6 +3516,64 @@ Future<Void> DDTeamCollection::serverTeamRemover() {
 	return DDTeamCollectionImpl::serverTeamRemover(this);
 }
 
+Future<Void> DDTeamCollection::fixWrongReplicasLoop() {
+	return DDTeamCollectionImpl::fixWrongReplicasLoop(this);
+}
+
+void DDTeamCollection::fixWrongReplicas() {
+	int maxTeamSize = maxLargeTeamSize();
+	int checkCount = 0;
+	for (auto& it : underReplication.ranges()) {
+		if (!it.value()) {
+			// The key range is not under-replicated
+			continue;
+		}
+		for (auto& r : shardsAffectedByTeamFailure->intersectingRanges(it.range())) {
+			if (++checkCount > SERVER_KNOBS->DD_MAXIMUM_LARGE_TEAM_CLEANUP) {
+				return;
+			}
+
+			auto& teams = r.value();
+			if (!teams.second.empty()) {
+				// The key range is currently being moved
+				continue;
+			}
+
+			int customReplicas = configuration.storageTeamSize;
+			for (auto& c : customReplication->intersectingRanges(r.range())) {
+				customReplicas = std::max(customReplicas, c.value());
+			}
+
+			int currentSize = 0;
+			for (auto& c : teams.first) {
+				if (c.primary == primary) {
+					currentSize = c.servers.size();
+				}
+			}
+
+			if (currentSize < customReplicas) {
+				// check if a larger team exists
+				if (maxTeamSize > currentSize) {
+					TraceEvent("FixWrongReplicas", distributorId)
```

Comment:
Same as before, might be useful to include desired replication (`customReplicas`) factor here.